packages/router/useRouteHash/index.test.ts

```diff
@@ -74,6 +74,28 @@ describe('useRouteHash', () => {
     expect(hash.value).toBe('foo')
   })
 
+  it('should trigger effects only once', async () => {
+    const route = getRoute()
+    const router = { replace: (r: any) => Object.assign(route, r) } as any
```

Comment:
This mock router implementation is different from other tests in this file, which use:

```js
const router = { replace: (r: any) => route = r } as any
```

Using basic assignment to replace the value of `route` (ie `route = r`) doesn't trigger watchers on `route`, so this mock behaves closer to the real vue-router.

I don't think it would be a bad idea to update the other tests in this file to use the same method, or even create a more realistic mock router that the tests can all use. However, I didn't feel that change would be in the scope of this PR.